.github/workflows/test-and-build.yml

```diff
@@ -0,0 +1,69 @@
+name: CI Pipeline
+
+on:
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - main
+  release:
+    types:
+      - published
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v2
+        with:
+          java-version: '17'
+          distribution: 'zulu'
+
+      - name: Run tests
+        run: mvn test
+
+  build:
+    runs-on: ubuntu-latest
+    needs: test
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v2
+        with:
+          java-version: '17'
+          distribution: 'zulu'
+
+      - name: Build project
+        run: mvn clean install
```

Comment:
why split test and install ? install made the test if you do not add skipTests

Reply:
The purpose of separating the test and build steps is to reduce the execution time of the pipeline, especially if the tests fail. This allows us to quickly identify any issues without waiting for the entire build phase to complete.

You're correct that it is also possible to skip the tests during the build phase, which can further optimize the process.